tapir/shifts/templates/shifts/shift_calendar_template.html

```diff
@@ -8,60 +8,78 @@
-        </div>
-        {% shift_filters %}
+    <div class="d-flex align-items-end flex-column m-2">
+        <button class="filter-button {% tapir_button_link %}"
+                data-bs-toggle="collapse"
+                data-bs-target="#collapseCalendarInfo"
+                aria-expanded="false"
+                style="text-decoration: none">
+            <span class="material-icons">unfold_more</span>{% translate "Legend & Filters" %}
+        </button>
     </div>
-    <div class="card mb-2">
-        <h5 class="card-header">{% translate "Date range" %}</h5>
-        <div class="card-body d-flex gap-4">
-            <script>
-                function onFilter() {
-                    let url = "{% url "shifts:calendar" %}";
-                    url += "?date_from=" + document.getElementById("date-from").value;
-                    url += "&date_to=" + document.getElementById("date-to").value;
-                    window.location = url;
-                }
-            </script>
-            <div class="d-flex align-items-baseline gap-2">
-                <label for="date-from" class="form-label">{% translate "From" %}</label>
-                <input id="date-from"
-                       type="date"
-                       value="{{ date_from }}"
-                       class="form-control"
-                       style="max-width: 160px" />
-            </div>
-            <div class="d-flex align-items-baseline gap-2">
-                <label for="date-to" class="form-label">{% translate "To" %}</label>
-                <input id="date-to"
-                       type="date"
-                       value="{{ date_to }}"
-                       class="form-control"
-                       style="max-width: 160px" />
+    <div class="collapse" id="collapseCalendarInfo">
+        <div class="card-group mb-2">
+            <div class="card">
+                <h5 class="card-header">
+                    {% block card_title %}
+                    {% endblock card_title %}
+                </h5>
+                <div class="card-body">
+                    {% block card_helptext %}
+                    {% endblock card_helptext %}
+                </div>
             </div>
-            <button class="{% tapir_button_link %}" onclick="onFilter()">{% translate "Filter" %}</button>
+            {% shift_filters %}
         </div>
     </div>
     {% for week, shifts_by_day in shifts_by_weeks_and_days.items %}
         <div class="card mb-2">
             {% with week_group=week_to_group|dictionary_get:week %}
-                <h5 class="card-header">
-                    KW {{ week|date:"W" }} {% translate 'starting' %} {{ week }},
-                    {% translate 'Week ' %}{{ week_group.name }}
-                </h5>
+                <div class="card-header">
+                    <div class="container-fluid">
+                        <div class="row">
+                            <div class="col-5 d-flex align-items-center">
```

Comment:
You should read what those classes are before copying them blind. See the bootstrap doc for layout/container, /grid and /columns: https://getbootstrap.com/docs/5.3/layout/containers/.

Remove the 2 divs with `container-fluid` and `row`, the inner div gets `class="col-5 d-flex flex-column justify-content-center"`, h5 gets `style="margin-bottom: 0"`. See this explanation of the flexbox css system: https://css-tricks.com/snippets/css/a-guide-to-flexbox/#flexbox-background

Reply:
The containers and grid are made to layout the page, for example to organize the cards or the sidebar, not for small elements like a title. That's why it doesn't work when aligning the title:
![image](https://github.com/user-attachments/assets/69002d90-c653-46a4-b82f-7af8d602a967)

Concerning the "container" that is named on the flexbox guide page, it's a name they used for their example. It is not the same class that bootstrap defines.

Concerning the solution I suggested, I forgot a wrapping div, sorry:
```html
<div class="card mb-2">
    {% with week_group=week_to_group|dictionary_get:week %}
         <div class="card-header">
            <div class="d-flex flex-row justify-content-between">
                <div class="d-flex flex-column justify-content-center">
                    <h5 style="margin-bottom: 0">
                        KW {{ week|date:"W" }} {% translate 'starting' %} {{ week }},
                        {% translate 'Week ' %}{{ week_group.name }}
                    </h5>
                </div>
         ....
```